File: misc/py/3827.count_monobit_integers [1].py
class Solution:
    # T.C.=O(log(n)), S.C.=O(1)
    # Math
    def countMonobit(self, n: int) -> int:
        return 1 + math.floor(math.log2(n+1))

# every monobit int is of the form (2^n - 1), n=0,1,2,...

File: CPP [3501-4000]/3827.count-monobit-integers [1].cpp
class Solution {
public:
    // T.C.=O(log(n)), S.C.=O(1)
    // Math
    int countMonobit(int n) {
        return 1 + floor(log2(n+1));
    }
};

// every monobit int is of the form (2^n - 1), n=0,1,2,...